src/main/kotlin/com/dhyer/bot_fighter/game_actions/MoveGameAction.kt

```diff
@@ -1,11 +1,27 @@
 package com.dhyer.bot_fighter.game_actions
 
+import com.dhyer.bot_fighter.Game
 import com.dhyer.bot_fighter.Player
+import java.awt.Point
 import java.time.LocalDateTime
 
-class MoveGameAction(player: Player, createdAt: LocalDateTime) : GameAction(player, createdAt) {
-  override fun execute() {
-    // TODO frd move players
-    println("\tMoving player")
+class MoveGameAction(player: Player, createdAt: LocalDateTime, val vector: Int) : GameAction(player, createdAt) {
+  override fun execute(opponent: Player) {
+    val x = getSingleXCoord(player)
+    val opponentX = getSingleXCoord(opponent)
+    if (x + vector < 0) {
+      println("\tCannot move into left wall")
+    } else if (x + vector > Game.WIDTH) {
+      println("\tCannot move into right wall")
+    } else if (x + vector == opponentX) {
+      println("\tCannot move into opponent")
+    } else {
+      player.location.forEach { it.x += vector }
+      println("\tMoving player ${this.player.id}")
+    }
+  }
+
+  fun getSingleXCoord(player: Player): Int {
+    return player.location.map { it.x }.min() ?: 0
```

Comment:
Why do we need the minimum x? Shouldn't they be equivalent in all cases aside from jump attack (which we shouldn't be invoking this action in that state)

Reply:
oob: don't necessarily need the minimum, was just used to grab the effective vertical space